service/frontend/workflowHandler.go

```diff
@@ -3011,37 +3018,594 @@ func (wh *WorkflowHandler) ListTaskQueuePartitions(ctx context.Context, request
+			Query: &querypb.WorkflowQuery{
+				QueryType: scheduler.QueryNameListMatchingTimes,
+				QueryArgs: queryPayload,
+			},
+		},
+	}
+	res, err := wh.historyClient.QueryWorkflow(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var response workflowservice.ListScheduleMatchingTimesResponse
+	err = payloads.Decode(res.GetResponse().GetQueryResult(), &response)
+	if err != nil {
+		return nil, err
+	}
+
+	return &response, nil
 }
 
 // Deletes a schedule, removing it from the system.
 func (wh *WorkflowHandler) DeleteSchedule(ctx context.Context, request *workflowservice.DeleteScheduleRequest) (_ *workflowservice.DeleteScheduleResponse, retError error) {
-	return nil, serviceerror.NewUnimplemented("todo")
+	defer log.CapturePanic(wh.logger, &retError)
+
+	if wh.isStopped() {
+		return nil, errShuttingDown
+	}
+
+	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+		return nil, err
+	}
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = wh.historyClient.TerminateWorkflowExecution(ctx, &historyservice.TerminateWorkflowExecutionRequest{
+		NamespaceId: namespaceID.String(),
+		TerminateRequest: &workflowservice.TerminateWorkflowExecutionRequest{
+			Namespace:         request.Namespace,
+			WorkflowExecution: &commonpb.WorkflowExecution{WorkflowId: request.ScheduleId},
+			Reason:            "terminated by DeleteSchedule",
+			Identity:          request.Identity,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &workflowservice.DeleteScheduleResponse{}, nil
 }
 
 // List all schedules in a namespace.
 func (wh *WorkflowHandler) ListSchedules(ctx context.Context, request *workflowservice.ListSchedulesRequest) (_ *workflowservice.ListSchedulesResponse, retError error) {
-	return nil, serviceerror.NewUnimplemented("todo")
+	defer log.CapturePanic(wh.logger, &retError)
+
+	if wh.isStopped() {
+		return nil, errShuttingDown
+	}
+
+	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+		return nil, err
+	}
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	if request.GetMaximumPageSize() <= 0 {
+		request.MaximumPageSize = int32(wh.config.VisibilityMaxPageSize(request.GetNamespace()))
+	}
+
+	if wh.isListRequestPageSizeTooLarge(request.GetMaximumPageSize(), request.GetNamespace()) {
+		return nil, serviceerror.NewInvalidArgument(fmt.Sprintf(errPageSizeTooBigMessage, wh.config.ESIndexMaxResultWindow()))
+	}
+
+	namespaceName := namespace.Name(request.GetNamespace())
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: should we respect this here?
+	if wh.config.DisableListVisibilityByFilter(namespaceName.String()) {
+		return nil, errNoPermission
+	}
```

Comment:
hmm, i think the reason we have that dynamic config is because the cassandra secondary index would bring down cass cluster if the data is too big. We probably should honor that setting. But maybe we should have clear error message or error logging to indicate that setting is the reason the request is failing. Otherwise, people won't be able to find out why it is failing. 
Later, we should really get rid of cassandra visibility store.

Reply:
I changed the error message to be more specific